### ISSUE TITLE
Inconsistencies in document-share

### ISSUE THREAD

**Issue by aleixpol** [NitruxSA/breeze-icon-theme#305]
There's different icons with different sizes for document-share. I was using it in QuickShare and now it looks a bit weird. I see 2 options:
* Get a just share icon.
* Keep document-share as just share.

Otherwise it feels like there's different meanings behind.
http://i.imgur.com/TkhcdSt.png

Thanks!

**Comment by UriHerrera** [comment]
The one largest one (the one with the paper sheet) is for 32px. And I'm sure it's meant to be used in LibreOffice.

**Comment by aleixpol** [comment]
Can't we just have 2 icons? one for share and another for document-share?

Otherwise we have inconsistencies: http://i.imgur.com/PnSF5YY.png (these 2 should be the same icon, and if you resize the panel the icon can change as well)

**Comment by UriHerrera** [comment]
We can, but in the cases where document-share (as in when not in the context of a document) share should be used and, share should be named share-something-something as to not have overly generic named icons.

**Comment by aleixpol** [comment]
What solution do you suggest?

**Comment by aleixpol** [comment]
Would it be possible then to have a generic "share" icon where the context isn't up to be specified to the actual icon? That is, for a generic tool that can share any kind of data, document or not.

**Comment by DarkknightAK** [comment]
we have action/document-share and apps/preferences-system-network-sharing. when you need an action icon use document-share and when you need an sharing application icon you can use preferences-system-network-sharing.

the document-share icon is quite generic you use the 32px icon but the 16 and 22px icon is really generic. https://github.com/NitruxSA/breeze-icon-theme/blob/master/Breeze/actions/22/document-share.svg

**Comment by DarkknightAK** [comment]
maybe if we remove the document at the 32px icon?

**Comment by aleixpol** [comment]
Thomas here:
Using different icons at different sizes to accommodate different uses of the same icon name is definitely not the way to go. This is what people did with Oxygen and it broke with the transition to Breeze. We need applications to use the semantically correct icon. 
This icon says "share-document" and therefore it does make sense that it has the document metaphor. But now Aleix is looking for an icon for something that can share data that is not documents, so he should get a new, semantically correct icon.

**Comment by DarkknightAK** [comment]
@ Thomas. That's how different sizes work. the icon is called document-share so in 22px you don't have space for the document but in 32px there is space. should we remove the document but it is called document-share.

@aleixpol that's easy please say for what do you need a new icon and we will add it. but without specification an generic generic icon isn't possible.

**Comment by aleixpol** [comment]
How about an "action-share" icon, if "share" isn't possible? Does that make sense?

**Comment by DarkknightAK** [comment]
and for what do you use it (screenshot and descripition please)

**Comment by aleixpol** [comment]
Quick Share plasmoid: http://i.imgur.com/YjTcv1S.png
KDevelop changes share button: http://i.imgur.com/u2u32Ht.png
Kamoso Share button: Doesn't exist yet, I don't have a screenshot

I don't think the document part adds anything to the plate, as it's about sharing information, regardless of how it's bundled. i.e. you can share text and images as well.

**Comment by aleixpol** [comment]
Invoke @colomar.

**Comment by colomar** [comment]
@DarkknightAK : Different sizes have different amount of details because on smaller sizes, too detailed icons don't work.
They should never change their semantics, though, because we can't guarantee that always the same size will be used. Plasmoids, for example, can have different sizes, so you can't say "Well, I'll use that icon which looks the way I want it to look at 22px", because when people resize it, it suddenly uses the 32px icon and if that has different semantics that don't work for the Plasmoid, it's broken.

Uri is right that applications should use the icon that is semantically right for them, not the one that looks fitting in a certain icon theme. People may change their icon theme, so if you choose an icon that works well in Breeze it may not work well in other themes.

**Comment by DarkknightAK** [comment]
@colomar they don't change there semantics
https://github.com/NitruxSA/breeze-icon-theme/blob/master/Breeze/actions/16/document-share.svg
https://github.com/NitruxSA/breeze-icon-theme/blob/master/Breeze/actions/22/document-share.svg
https://github.com/NitruxSA/breeze-icon-theme/blob/master/Breeze/actions/32/document-share.svg
all say share and the icon name is also clear document-share

**Comment by colomar** [comment]
@DarkknightAK I agree they don't. I was just arguing against your suggestion to simply remove the document metaphor from the 32px icon.
We absolutely agree: The icon document-share simply is the wrong icon for the purpose Aleix needs an icon for.

He needs an icon used for sharing anything, not just documents. But since you're already working with him to create such an icon, I'll shut up ;)
